lib/tap.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+var format = require('util').format;
+
+// Parses stack trace and extracts original function name, file name and line.
+function getSourceFromStack(stack, index) {
+	return stack
+		.split('\n')
+		.slice(index, index + 1)
+		.join('')
+		.replace(/^\s+at /, '');
+}
+
+exports.start = function () {
+	console.log('TAP version 13');
+};
+
+var i = 0;
+
+exports.test = function (test) {
+	var output = [];
+
+	output.push('# ' + test.title);
+
+	if (test.error) {
+		output.push(format('not ok %d - %s', ++i, test.error.message));
+		output.push('  ---');
+		output.push('    operator: ' + test.error.operator);
+		output.push('    expected: ' + test.error.expected);
+		output.push('    actual: ' + test.error.actual);
+		output.push('    at: ' + getSourceFromStack(test.error.stack, 3));
+		output.push('  ...');
+	} else {
+		output.push(format('ok %d - %s', ++i, test.title));
+	}
+
+	console.log(output.join('\n'));
+};
+
+exports.unhandledError = function (err) {
+	var output = [];
+
+	output.push('# ' + err.message);
+	output.push(format('not ok %d - %s', ++i, err.message));
+	output.push('  ---');
+	output.push('    name: ' + err.name);
+	output.push('    at: ' + getSourceFromStack(err.stack, 1));
+	output.push('  ...');
+
+	console.log(output.join('\n'));
+};
```

Comment:
I think something like this would look nicer:

```js
console.log([
	'# ' + err.message,
	format('not ok %d - %s', ++i, err.message),
	'  ---',
	'    name: ' + err.name,
	'    at: ' + getSourceFromStack(err.stack, 1),
	'  ...'
].join('\n'));
```